### ISSUE TITLE
Misc. bug: The new version has become very slow in inference on the CPU

### ISSUE THREAD

**Issue by hanyoud** [ggml-org/llama.cpp#18592]
### Name and Version

version: 7520 (12ee1763a)
built with GNU 11.4.0 for Linux x86_64

### Operating systems

Linux

### Which llama.cpp modules do you know to be affected?

llama-cli

### Command line

```shell
bin/llama-cli -m ../models/HY-MT1.5-7B-Q8_0.gguf -p "将以下文本翻译为英文，注意只需 要输出翻译后的结果，不要额外解释：\n\n校长说衣服上除了校徽别别别的你偏要别别的。" -n 4096 --temp 0.7 --top-k 20 --top-p 0.6 --repeat-penalty 1.05 --no-warmup -t 10
```

### Problem description & steps to reproduce

I got 0.2 t/s on AMD Ryzen AI 9 H 365 w/ Radeon 880M with llama.cpp-b7520, and i got 10 t/s with llama.cpp-b6981.

➜  llama.cpp-b7520-bin$ bin/llama-cli -m ../models/HY-MT1.5-7B-Q8_0.gguf -p "将以下文本翻译为英文，注意只需 要输出翻译后的结果，不要额外解释：\n\n校长说衣服上除了校徽别别别的你偏要别别的。" -n 4096 --temp 0.7 --top-k 20 --top-p 0.6 --repeat-penalty 1.05 --no-warmup -t 10
load_backend: loaded RPC backend from /home/hanyoud/ai/llama.cpp-b7520-bin/bin/libggml-rpc.so
load_backend: loaded CPU backend from /home/hanyoud/ai/llama.cpp-b7520-bin/bin/libggml-cpu-icelake.so

Loading model...  

▄▄ ▄▄
██ ██
██ ██  ▀▀█▄ ███▄███▄  ▀▀█▄    ▄████ ████▄ ████▄
██ ██ ▄█▀██ ██ ██ ██ ▄█▀██    ██    ██ ██ ██ ██
██ ██ ▀█▄██ ██ ██ ██ ▀█▄██ ██ ▀████ ████▀ ████▀
                                    ██    ██
                                    ▀▀    ▀▀

build      : b7520-12ee1763a
model      : HY-MT1.5-7B-Q8_0.gguf
modalities : text

available commands:
  /exit or Ctrl+C     stop or exit
  /regen              regenerate the last response
  /clear              clear the chat history
  /read               add a text file

> 将以下文本翻译为英文，注意只需要输出翻译后的结果，不要额外解释：

校长说衣服上除了校徽别别别的你偏要别别的。

The principal said that besides the school emblem,

[ Prompt: 7.7 t/s | Generation: 0.2 t/s ]

### Relevant log output

➜  llama.cpp-b6981-bin $ bin/llama-cli -m ../models/HY-MT1.5-7B-Q8_0.gguf -p "将以下文本翻译为英文，注意只需 要输出翻译后的结果，不要额外解释：\n\n校长说衣服上除了校徽别别别的你偏要别别的。" -n 4096 --temp 0.7 --top-k 20 --top-p 0.6 --repeat-penalty 1.05 --no-warmup -t 10
load_backend: loaded RPC backend from /home/hanyoud/ai/llama.cpp-b6981-bin/bin/libggml-rpc.so
load_backend: loaded CPU backend from /home/hanyoud/ai/llama.cpp-b6981-bin/bin/libggml-cpu-icelake.so
build: 6981 (647b960bd) with cc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0 for x86_64-linux-gnu
main: llama backend init
main: load the model and apply lora adapter, if any
llama_model_loader: loaded meta data with 35 key-value pairs and 354 tensors from ../models/HY-MT1.5-7B-Q8_0.gguf (version GGUF V3 (latest))
llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.
llama_model_loader: - kv   0:                       general.architecture str              = hunyuan-dense
llama_model_loader: - kv   1:                               general.type str              = model
llama_model_loader: - kv   2:                     general.sampling.top_k i32              = 20
llama_model_loader: - kv   3:                     general.sampling.top_p f32              = 0.800000
llama_model_loader: - kv   4:                      general.sampling.temp f32              = 0.700000
llama_model_loader: - kv   5:                               general.name str              = HY MT1.5 7B
llama_model_loader: - kv   6:                           general.basename str              = HY-MT1.5
llama_model_loader: - kv   7:                         general.size_label str              = 7B
llama_model_loader: - kv   8:                               general.tags arr[str,1]       = ["translation"]
llama_model_loader: - kv   9:                          general.languages arr[str,36]      = ["zh", "en", "fr", "pt", "es", "ja", ...
llama_model_loader: - kv  10:                  hunyuan-dense.block_count u32              = 32
llama_model_loader: - kv  11:               hunyuan-dense.context_length u32              = 262144
llama_model_loader: - kv  12:             hunyuan-dense.embedding_length u32              = 4096
llama_model_loader: - kv  13:          hunyuan-dense.feed_forward_length u32              = 14336
llama_model_loader: - kv  14:         hunyuan-dense.attention.head_count u32              = 32
llama_model_loader: - kv  15:      hunyuan-dense.attention.head_count_kv u32              = 8
llama_model_loader: - kv  16:               hunyuan-dense.rope.freq_base f32              = 11158840.000000
llama_model_loader: - kv  17: hunyuan-dense.attention.layer_norm_rms_epsilon f32              = 0.000010
llama_model_loader: - kv  18:         hunyuan-dense.attention.key_length u32              = 128
llama_model_loader: - kv  19:       hunyuan-dense.attention.value_length u32              = 128
llama_model_loader: - kv  20:            hunyuan-dense.rope.scaling.type str              = none
llama_model_loader: - kv  21:          hunyuan-dense.rope.scaling.factor f32              = 1.000000
llama_model_loader: - kv  22: hunyuan-dense.rope.scaling.original_context_length u32              = 262144
llama_model_loader: - kv  23:                       tokenizer.ggml.model str              = gpt2
llama_model_loader: - kv  24:                         tokenizer.ggml.pre str              = hunyuan
llama_model_loader: - kv  25:                      tokenizer.ggml.tokens arr[str,128167]  = ["!", "\"", "#", "$", "%", "&", "'", ...
llama_model_loader: - kv  26:                  tokenizer.ggml.token_type arr[i32,128167]  = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...
llama_model_loader: - kv  27:                      tokenizer.ggml.merges arr[str,264306]  = ["Ġ Ġ", "Ġ ĠĠĠ", "ĠĠ ĠĠ", "...
llama_model_loader: - kv  28:                tokenizer.ggml.bos_token_id u32              = 127958
llama_model_loader: - kv  29:                tokenizer.ggml.eos_token_id u32              = 127960
llama_model_loader: - kv  30:            tokenizer.ggml.padding_token_id u32              = 127961
llama_model_loader: - kv  31:          tokenizer.ggml.seperator_token_id u32              = 127962
llama_model_loader: - kv  32:                    tokenizer.chat_template str              = {% set ns = namespace(has_head=true) ...
llama_model_loader: - kv  33:               general.quantization_version u32              = 2
llama_model_loader: - kv  34:                          general.file_type u32              = 7
llama_model_loader: - type  f32:  129 tensors
llama_model_loader: - type q8_0:  225 tensors
print_info: file format = GGUF V3 (latest)
print_info: file type   = Q8_0
print_info: file size   = 7.43 GiB (8.50 BPW) 
load: special_eos_id is not in special_eog_ids - the tokenizer config may be incorrect
load: printing all EOG tokens:
load:   - 127957 ('<|endoftext|>')
load:   - 127960 ('<|eos|>')
load: special tokens cache size = 209
load: token to piece cache size = 0.7868 MB
print_info: arch             = hunyuan-dense
print_info: vocab_only       = 0
print_info: n_ctx_train      = 262144
print_info: n_embd           = 4096
print_info: n_embd_inp       = 4096
print_info: n_layer          = 32
print_info: n_head           = 32
print_info: n_head_kv        = 8
print_info: n_rot            = 128
print_info: n_swa            = 0
print_info: is_swa_any       = 0
print_info: n_embd_head_k    = 128
print_info: n_embd_head_v    = 128
print_info: n_gqa            = 4
print_info: n_embd_k_gqa     = 1024
print_info: n_embd_v_gqa     = 1024
print_info: f_norm_eps       = 0.0e+00
print_info: f_norm_rms_eps   = 1.0e-05
print_info: f_clamp_kqv      = 0.0e+00
print_info: f_max_alibi_bias = 0.0e+00
print_info: f_logit_scale    = 0.0e+00
print_info: f_attn_scale     = 0.0e+00
print_info: n_ff             = 14336
print_info: n_expert         = 0
print_info: n_expert_used    = 0
print_info: n_expert_groups  = 0
print_info: n_group_used     = 0
print_info: causal attn      = 1
print_info: pooling type     = 0
print_info: rope type        = 2
print_info: rope scaling     = none
print_info: freq_base_train  = 11158840.0
print_info: freq_scale_train = 1
print_info: n_ctx_orig_yarn  = 262144
print_info: rope_finetuned   = unknown
print_info: model type       = 7B
print_info: model params     = 7.50 B
print_info: general.name     = HY MT1.5 7B
print_info: vocab type       = BPE
print_info: n_vocab          = 128167
print_info: n_merges         = 264306
print_info: BOS token        = 127958 '<|startoftext|>'
print_info: EOS token        = 127960 '<|eos|>'
print_info: EOT token        = 127957 '<|endoftext|>'
print_info: SEP token        = 127962 '<|extra_0|>'
print_info: PAD token        = 127961 '<|pad|>'
print_info: LF token         = 198 'Ċ'
print_info: EOG token        = 127957 '<|endoftext|>'
print_info: EOG token        = 127960 '<|eos|>'
print_info: max token length = 256
load_tensors: loading model tensors, this can take a while... (mmap = true)
load_tensors: offloading 32 repeating layers to GPU
load_tensors: offloading output layer to GPU
load_tensors: offloaded 33/33 layers to GPU
load_tensors:   CPU_Mapped model buffer size =  7604.99 MiB
...............................................................................................
llama_context: constructing llama_context
llama_context: n_seq_max     = 1
llama_context: n_ctx         = 4096
llama_context: n_ctx_seq     = 4096
llama_context: n_batch       = 2048
llama_context: n_ubatch      = 512
llama_context: causal_attn   = 1
llama_context: flash_attn    = auto
llama_context: kv_unified    = false
llama_context: freq_base     = 11158840.0
llama_context: freq_scale    = 1
llama_context: n_ctx_seq (4096) < n_ctx_train (262144) -- the full capacity of the model will not be utilized
llama_context:        CPU  output buffer size =     0.49 MiB
llama_kv_cache:        CPU KV buffer size =   512.00 MiB
llama_kv_cache: size =  512.00 MiB (  4096 cells,  32 layers,  1/1 seqs), K (f16):  256.00 MiB, V (f16):  256.00 MiB
llama_context: Flash Attention was auto, set to enabled
llama_context:        CPU compute buffer size =   258.33 MiB
llama_context: graph nodes  = 1127
llama_context: graph splits = 1
common_init_from_params: added <|endoftext|> logit bias = -inf
common_init_from_params: added <|eos|> logit bias = -inf
common_init_from_params: setting dry_penalty_last_n to ctx_size = 4096
main: llama threadpool init, n_threads = 10
main: chat template is available, enabling conversation mode (disable it with -no-cnv)
*** User-specified prompt will pre-start conversation, did you mean to set --system-prompt (-sys) instead?
main: chat template example:
<|startoftext|>You are a helpful assistant<|extra_4|><|startoftext|>Hello<|extra_0|>Hi there<|eos|><|startoftext|>How are you?<|extra_0|>

system_info: n_threads = 10 (n_threads_batch = 10) / 20 | CPU : SSE3 = 1 | SSSE3 = 1 | AVX = 1 | AVX2 = 1 | F16C = 1 | FMA = 1 | BMI2 = 1 | AVX512 = 1 | AVX512_VBMI = 1 | AVX512_VNNI = 1 | LLAMAFILE = 1 | OPENMP = 1 | REPACK = 1 | 

main: interactive mode on.
sampler seed: 1232299469
sampler params: 
	repeat_last_n = 64, repeat_penalty = 1.050, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = 4096
	top_k = 20, top_p = 0.600, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.700
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
sampler chain: logits -> logit-bias -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
generate: n_ctx = 4096, n_batch = 2048, n_predict = 4096, n_keep = 0

== Running in interactive mode. ==
 - Press Ctrl+C to interject at any time.
 - Press Return to return control to the AI.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.
 - Not using system message. To change it, set a different value via -sys PROMPT

将以下文本翻译为英文，注意只需要输出翻译后的结果，不要额外解释：

校长说衣服上除了校徽别别别的你偏要别别的。The principal said that apart from the school emblem, there shouldn’t be anything else on the clothes, but you insisted on adding something else anyway.

> 
llama_perf_sampler_print:    sampling time =      12.40 ms /    66 runs   (    0.19 ms per token,  5324.30 tokens per second)
llama_perf_context_print:        load time =    3811.10 ms
llama_perf_context_print: prompt eval time =     444.24 ms /    36 tokens (   12.34 ms per token,    81.04 tokens per second)
llama_perf_context_print:        eval time =    2984.14 ms /    29 runs   (  102.90 ms per token,     9.72 tokens per second)
llama_perf_context_print:       total time =   14992.36 ms /    65 tokens
llama_perf_context_print:    graphs reused =         28
llama_memory_breakdown_print: | memory breakdown [MiB] | total   free    self   model   context   compute    unaccounted |
llama_memory_breakdown_print: |   - Host               |                 8375 =  7604 +     512 +     258                |

**Comment by CISC** [comment]
Since you're comparing against `b6981`, try using `llama-completion` instead.

**Comment by hanyoud** [comment]
> Since you're comparing against `b6981`, try using `llama-completion` instead.

thx.I have re-run with `llama-completion`,and got some infos: `b7520` use `-c,    --ctx-size N` with default 0, while the default value is 4096 in `b6981`.
